src/core/room.ts

```diff
@@ -188,9 +189,19 @@ export class DefaultRoom implements Room {
     // Room initialization: handle the state before features start to be initialized
     const rejectablePromise = makeRejectablePromise(makeAlwaysResolves(initAfter));
     const initFeaturesAfter = rejectablePromise.promise;
+    // Catch to prevent global errors and print a debug log
```

Comment:
This one might not be needed because we always create `_asyncOpsAfter`, which is using `initFeaturesAfter.then(...)` and it is created sync in this constructor, so before `initFeaturesAfter` can be rejected.

`_asyncOpsAfter` has a catch already, at the bottom of the constructor.

Reply:
We'll still need it.

The `initFeaturesAfter.then` and `initFeaturesAfter.catch` calls returns a new promise that executes based on the result of `initFeaturesAfter`.

So the call to `initFeaturesAfter.catch` here is one promise chain, `_asyncOpsAfter` is another separate one. Both of them will reject if `_initFeaturesAfter` rejects.

If we wanted to not have this behaviour, then `initFeaturesAfter` would have to be set to the value of `initFeaturesAfter.catch(...)` - which I don't think is desirable.

Reply:
Yeah - I've removed it and left a comment to that effect.